lib/operations/mongo_client_ops.js

```diff
@@ -194,6 +197,31 @@ function connect(mongoClient, url, options, callback) {
       delete _finalOptions.db_options.auth;
     }
 
+    // APM logging
+    if (_finalOptions.monitor) {
+      // get array of monitor groups to enable
+      const monitors = getMonitors(_finalOptions.monitor);
+      // check if logging enables command monitoring
+      if (monitors.indexOf('command') > -1) _finalOptions.monitorCommands = true;
+
+      const defaultFile = process.cwd() + '/monitor.log';
+      // get writable out stream, either to custom location or default
+      const out = _finalOptions.monitorout
+        ? getCustomOut(_finalOptions.monitorout)
```

Comment:
`getCustomOut` seems to resolve a given path, and then create a writable stream to it. It seems like we should maybe have something more like `const monitoringStream = createMonitoringStream(resolveMonitoringOutputFile(_finalOptions.monitorOut))`. This way we can clearly separate the purposes of each function, and convey intent with naming.

Reply:
Also, this makes it very clear (in a single function) all the assumptions you make about e.g. file naming.

Reply:
is it ok if `resolveMonitoringOutputFile` can return `stdout` or `stderr`, since those aren't conventional files?

Reply:
yeah just the special case strings `stderr`/`stdout` seems reasonable to me